src/evm/protocol/uniswap_v4/hooks/generic_vm_hook_handler.rs

```diff
@@ -269,4 +309,64 @@ mod tests {
         );
         assert_eq!(res.fee, U24::from(0));
     }
+
+    #[test]
+    fn test_after_swap() {
+        let block = BlockHeader {
+            number: 15797251,
+            hash: B256::from_str(
+                "0x7032b93c5b0d419f2001f7c77c19ade6da92d2df147712eac1a27c7ffedfe410",
+            )
+            .unwrap(),
+            timestamp: 1748397011,
+        };
+        let db = SimulationDB::new(get_client(None), get_runtime(), Some(block));
+        let engine = create_engine(db, true).expect("Failed to create simulation engine");
+
+        // pool manager on ethereum
+        let pool_manager = Address::from_str("0x000000000004444c5dc75cB358380D2e3dE08A90")
+            .expect("Invalid pool manager address");
+
+        let hook_address = Address::from_str("0x0010d0d5db05933fa0d9f7038d365e1541a41888")
+            .expect("Invalid hook address");
+
+        // This bytecode corresponds to LimitOrder example hook https://github.com/Uniswap/v4-periphery/blob/example-contracts/contracts/hooks/examples/LimitOrder.sol
+        // To get the bytecode:
+        //  - we had to change the `IPoolManager public immutable manager` from an immutable to a
+        //    constant and hardcode the pool manager address. This is necessary because the
+        //    immutable variables are only filled in at (real) deployment time, if we just want to
+        //    inspect the bytecode they will be set to zero.
+        //  - `forge inspect LimitOrder deployedBytecode` to get the bytecode.
+        let bytecode =
+            Bytecode::new_raw(include_bytes!("assets/after_swap_test_hook_bytecode.bin").into());
+
+        let hook_handler = GenericVMHookHandler::new(hook_address, bytecode, engine, pool_manager)
+            .expect("Failed to create GenericVMHookHandler");
+
+        let context = StateContext {
+            currency_0: Address::from_str("0x0000000000000000000000000000000000000000").unwrap(),
+            currency_1: Address::from_str("0x000000c396558ffbab5ea628f39658bdf61345b3").unwrap(),
+            fees: UniswapV4Fees { zero_for_one: 0, one_for_zero: 0, lp_fee: 1 },
+            tick: 60,
+        };
+        let swap_params = SwapParams {
+            zero_for_one: true,
+            amount_specified: I256::try_from(-200000000000000000i128).unwrap(),
+            sqrt_price_limit: U256::from(4295128740u64),
+        };
+
+        let after_swap_params = AfterSwapParameters {
+            context,
+            sender: Address::from_str("0x66a9893cc07d91d95644aedd05d03f95e1dba8af").unwrap(),
+            swap_params,
+            delta: I256::from_dec_str("-3777134272822416944443458142492627143113384069767150805")
+                .unwrap(),
+            hook_data: Bytes::new(),
+        };
+
+        let result = hook_handler.after_swap(after_swap_params, block.number);
+
+        let res = result.unwrap().result;
+        assert_eq!(res, I256::from_raw(U256::from_str("0").unwrap()));
```

Comment:
Which transaction is this simulating? Can you include this here? Otherwise how do we know this value is right?